lib/msal-node/src/network/HttpClient.ts

```diff
@@ -60,29 +59,188 @@ export class HttpClient implements INetworkModule {
         options?: NetworkRequestOptions,
         cancellationToken?: number,
     ): Promise<NetworkResponse<T>> {
-        const request: AxiosRequestConfig = {
-            method: HttpMethod.POST,
-            url: url,
-            /* istanbul ignore next */
-            data: (options && options.body) || Constants.EMPTY_STRING,
+        if (!options?.proxyUrl) {
+            return networkRequestViaHttps(url, HttpMethod.POST, options, cancellationToken);
+        }
+
+        const destinationUrl = new URL(url);
+        const proxyUrl = new URL(options.proxyUrl);
+
+        const tunnelRequestOptions = {
+            host: proxyUrl.hostname,
+            port: proxyUrl.port,
+            method: "CONNECT",
+            path: destinationUrl.hostname,
+            headers: options?.headers || {},
             timeout: cancellationToken,
-            /* istanbul ignore next */
-            headers: options && options.headers,
-            /* istanbul ignore next */
-            validateStatus: () => true
         };
 
-        if (options && options.proxyUrl) {
-            // for axios, this has to be disabled
-            request.proxy = false;
-            request.httpsAgent = createHttpsProxyAgent(options.proxyUrl);
+        const objString = options?.body || "";
+        const outgoingRequestString = `${HttpMethod.POST.toUpperCase()} ${destinationUrl.href} HTTP/1.1\r\n` +
+            `Host: ${destinationUrl.host}\r\n` +
+            "Content-Type: application/x-www-form-urlencoded\r\n" +
+            `Content-Length: ${objString.length}\r\n` +
+            "Connection: close\r\n" +
+            `\r\n${objString}\r\n`;
```

Comment:
Can the new code in this file be factored into helper functions? Looks like some duplicated logic.

Reply:
In fact, can you factor the proxy calls into their own function, as you have done with the non proxy calls?

Reply:
I've already made helper functions. Are you saying you would like me to put the entirety of the logic into the helper functions? The way it's set up now: I only need to pass in 2 params instead of a bunch of other ones if I move the string builder into the helper function.

Reply:
Can there be a common util API for the string builder logic (POST will have extra changes)? More for code clarity

Reply:
Lets not worry about it for now, can be addressed incrementally.